Kira_iris_regression.R

```diff
@@ -0,0 +1,74 @@
+iris	# Fisherʻs iris dataset
+dim(iris)	# dimensions
+names(iris)
+
+## Renaming iris with more convenient names
+## Accessing vectors within iris, saving as separate objects
+
+names(iris) <- c("sepalL", "sepalW",  "petalL", "petalW", "species")
+species <- iris$species   
+x <- iris$petalL
+y <- iris$petalW
+
+
+## Illustrating regression by graphing 
+
+fit <- lm( y ~ x )  # fitting the linear model
+b <- coef( fit ) # grabbing the coefficients
+int <- b[1] # intercept
+slope <- b[2] # slope
+
+plot(x, y) # plotting the data points
+abline(fit) # drawing the regression line on the plot
+# look up help for ?abline and figure out how to use 
+# abline by specify slope and intercept.
+
+yhat <- slope*x + int # the fitted values
+
+points( x, yhat, pch=3)  # adding the fitted points
+
+## if we want to draw line segments between the 150th
+## point and the fitted regression value for it:
+i=150
+segments( x[i], y[i], x[i], yhat[i], col="red" )
+
+## R is "vectorized": it will work on vectors all at once:
+segments( x, y, x, yhat, col="red", lty=2 )
+
+sse <- (yhat-y)
+SSE <- sum((sse))^2
+
+text(2,2.5, paste("Slope:", slope))
+text(2,2.25, paste("y-int:", int))
+text(2,2, paste("SSE:", SSE))
+
+
+
+#Function Wrap
+myplot <- function(x=iris$petalL, y=iris$petalW, myint=int, myslope=slope)	{
+	yhat <- myslope*x + myint
+	se <- yhat - y
+	sse <- sum(se)^2
+	plot(x, y)
+	abline(myint, myslope)
+	points(x, yhat, pch=3)
+	segments(x, y, x, yhat, col="red", lty=2)
+	text(2,2.5, paste("Slope:", myslope))
+	text(2,2.25, paste("y-int:", myint))
+	text(2,2, paste("SSE:", sse))
+}
+
+myplot()
+myplot(myint=4, myslope=1)
+myplot(myint=1, myslope=4)
+myplot(myint=0.5, myslope=0.5)
+myplot(myint=5, myslope=5)
+myplot(myint=0, myslope=1)
+
+
+#My Function
+carsfunction <- function(x=cars$speed, y=cars$dist)	{
+	x <- x*1.60934
+	y <- y*1.60934
```

Comment:
Nice function, I think it would be good to have a title that is more informative (maybe milestokilometer?)

Reply:
Just changed it! Good idea